cartography/intel/aws/identitycenter.py

```diff
@@ -411,66 +443,108 @@ def get_user_group_memberships(
 
 
 @timeit
-def get_permset_roles(
+def _get_permset_roles(
     neo4j_session: neo4j.Session,
-    role_assignments: List[Dict[str, Any]],
-) -> List[Dict[str, Any]]:
+    permset_ids: list[str],
+) -> dict[tuple[str, str], str]:
     """
-    Enrich role assignments with exact role ARNs by querying existing permission set relationships.
-    Uses the ASSIGNED_TO_ROLE relationships created when permission sets were loaded.
+    Given a list of permission set ARNs, return a mapping of (permission set ARN, account ID) to role ARN
+    based on the ASSIGNED_TO_ROLE relationship in the graph.
     """
-    # Get unique permission set ARNs from role assignments
-    permset_ids = list({ra["PermissionSetArn"] for ra in role_assignments})
-
     query = """
     MATCH (role:AWSRole)<-[:ASSIGNED_TO_ROLE]-(permset:AWSPermissionSet)
+    MATCH (account:AWSAccount)-[:RESOURCE]->(role)
     WHERE permset.arn IN $PermSetIds
-    RETURN permset.arn AS PermissionSetArn, role.arn AS RoleArn
+    RETURN permset.arn AS PermissionSetArn, role.arn AS RoleArn, account.id AS AccountId
     """
     permset_to_role = neo4j_session.execute_read(
         read_list_of_dicts_tx,
         query,
         PermSetIds=permset_ids,
     )
-
-    # Create mapping from permission set ARN to role ARN
-    permset_to_role_map = {
```

Comment:
This is (one of) the main bug(s). A permset can map to more than one role. The way this is written, we map to just one!